tests/ui/manual_ignore_case_cmp.rs

```diff
@@ -0,0 +1,71 @@
+#![allow(clippy::all)]
+#![deny(clippy::manual_ignore_case_cmp)]
+
+use std::ffi::{OsStr, OsString};
+
+fn main() {}
+
+fn variants(a: &str, b: &str) -> bool {
+    if a.to_ascii_lowercase() == b.to_ascii_lowercase() {
+        return true;
+    }
+    if a.to_ascii_uppercase() == b.to_ascii_uppercase() {
+        return true;
+    }
+    let r = a.to_ascii_lowercase() == b.to_ascii_lowercase();
+    let r = r || a.to_ascii_uppercase() == b.to_ascii_uppercase();
+    r && a.to_ascii_lowercase() == b.to_uppercase().to_ascii_lowercase()
+}
+
+fn unsupported(a: char, b: char) {
+    // TODO:: these are rare, and might not be worth supporting
+    a.to_ascii_lowercase() == char::to_ascii_lowercase(&b);
+    char::to_ascii_lowercase(&a) == b.to_ascii_lowercase();
+    char::to_ascii_lowercase(&a) == char::to_ascii_lowercase(&b);
+}
+
+fn simple_char(a: char, b: char) {
+    a.to_ascii_lowercase() == b.to_ascii_lowercase();
+    a.to_ascii_lowercase() == *&b.to_ascii_lowercase();
+    *&a.to_ascii_lowercase() == b.to_ascii_lowercase();
+}
+fn simple_u8(a: u8, b: u8) {
+    a.to_ascii_lowercase() == b.to_ascii_lowercase();
+}
+fn simple_str(a: &str, b: &str) {
+    a.to_ascii_lowercase() == b.to_ascii_lowercase();
+    a.to_uppercase().to_ascii_lowercase() == b.to_ascii_lowercase();
+}
+fn simple_string(a: String, b: String) {
+    a.to_ascii_lowercase() == b.to_ascii_lowercase();
+}
+fn simple_string2(a: String, b: &String) {
+    a.to_ascii_lowercase() == b.to_ascii_lowercase();
```

Comment:
It would be good to have some tests with `!=` and ones where only one side is converted etc.

Reply:
thx, fixed